packages/samples/mvp-dapp/src/actors/omnilock.model.ts

```diff
@@ -17,10 +32,11 @@ export class OmnilockModel extends JSONStore<Record<string, never>> {
       schemaPattern?: SchemaPattern
     },
   ) {
-    super(undefined, params)
+    super(undefined, { ...params, ref: ActorReference.newWithPattern(OmnilockModel, `/${args}/`) })
```

Comment:
How about set `ref` in `register`, but not by construct?
```typescript
      this.#container
        .bind(ref.uri)
        .toDynamicValue(() => {
             if (module) {
const moduleInstance = new module(
                ...(params as ActorParamType[]).map((param) => param.routerParam).map((param) => ref.params[param]),
              )
moduleInstance.ref = ref
return moduleInstance
             }
         },
        )
        .inSingletonScope()
```

Reply:
> How about set `ref` in `register`, but not by construct?
> 
> ```ts
>       this.#container
>         .bind(ref.uri)
>         .toDynamicValue(() => {
>              if (module) {
> const moduleInstance = new module(
>                 ...(params as ActorParamType[]).map((param) => param.routerParam).map((param) => ref.params[param]),
>               )
> moduleInstance.ref = ref
> return moduleInstance
>              }
>          },
>         )
>         .inSingletonScope()
> ```

I designed to use the `ref` to save parameters, which are used in constructor of `Store`. So set the `ref` after the initiation might be useless